funding/forms.py

```diff
@@ -0,0 +1,115 @@
+from django import forms
+from django.utils.translation import gettext_lazy as _
+from .models import FundingSource
+from .models import Publication
+
+from institution.models import Institution
+
+
+class FundingSourceForm(forms.ModelForm):
+    class Meta:
+        model = FundingSource
+        fields = ['title', 'identifier', 'amount', 'funding_body', 'pi_email']
+
+    def __init__(self, user, *args, **kwargs):
+        instance = kwargs.get('instance', {})
+        if hasattr(instance, 'pi') and instance.pi is not None:
+            initial = kwargs.get('initial', {})
+            initial['pi_email'] = instance.pi.email
+            kwargs['initial'] = initial
+        super(FundingSourceForm, self).__init__(*args, **kwargs)
+        if user.profile.institution is not None:
+            self.fields['pi_email'].label = _(
+                '{} Principal Investigator Email address'.format(
+                    user.profile.institution.name
+                )
+            )
+            if user.profile.institution.funding_database_name != '':
+                self.fields['identifier'].label = _(
+                    '{} identifier'.format(
+                        user.profile.institution.funding_database_name
+                    )
+                )
+        # Set the initial email if pi is a user
+        self.user = user
+        self.user_email = user.email
+
+    def clean_pi_email(self):
+        cleaned_data = super().clean()
+        email = cleaned_data['pi_email']
```

Comment:
Could also use `.exists()`, to check if the domain is valid.

https://docs.djangoproject.com/en/2.1/ref/models/querysets/#exists